instrumentation/redis/test/opentelemetry/instrumentation/redis/patches/client_test.rb

```diff
@@ -201,6 +201,72 @@ def redis_with_auth(redis_options = {})
       _(last_span.attributes['net.peer.port']).must_equal redis_port
     end
 
+    it 'records nil' do
+      redis = redis_with_auth
+      redis.set('K', nil)
+
+      _(last_span.name).must_equal 'SET'
+      _(last_span.attributes['db.statement']).must_equal 'SET K '
+    end
+
+    it 'records empty string' do
+      redis = redis_with_auth
+      redis.set('K', '')
+
+      _(last_span.name).must_equal 'SET'
+      _(last_span.attributes['db.statement']).must_equal 'SET K '
+    end
+
+    it 'truncates long command values' do
+      redis = redis_with_auth
+      the_long_value = 'y' * 100
+
+      redis.set('K', the_long_value)
+      _(last_span.name).must_equal 'SET'
+      _(last_span.attributes['db.statement']).must_equal 'SET K ' + 'y' * 47 + '...'
+    end
+
+    it 'truncates long db.statements' do
+      redis = redis_with_auth
+      the_long_value = 'y' * 100
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.queue([:set, 'v1', the_long_value])
+      redis.commit
+
+      expected_db_statement = <<~HEREDOC.chomp
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy...
+        SET v1 yyyyyyyyyyyyyyyyyyyyyyyyyy...
+      HEREDOC
+
+      _(last_span.name).must_equal 'SET SET SET SET SET SET SET SET SET'
```

Comment:
I really think we could adhere to the spec and also avoid having span names like this, IMO it makes more sense to call these spans `pipelined` or something very similar to this.

Reply:
If I saw a span name like this, my first thought would be "gee, there's a bug in the instrumentation"

Reply:
Agreed - let's use `pipelined`. The `db.statement` will have the full query.